src/NUnitFramework/framework/Internal/TestFilter.cs

```diff
@@ -62,9 +62,13 @@ public bool IsEmpty
         /// to change the behavior of the filter.
         /// </summary>
         /// <param name="test">The test to which the filter is applied</param>
+        /// <param name="negated">If set to <c>true</c> we are carrying a negation through</param>
         /// <returns>True if the test passes the filter, otherwise false</returns>
-        public virtual bool Pass(ITest test)
+        public virtual bool Pass(ITest test, bool negated = false)
```

Comment:
Could we revert the change to the interface and the change to this method and add a new virtual `bool Pass(ITest, bool)` method? That would enable libraries that call the existing methods to continue working against both past and future versions of NUnit at the same time.

https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md#members

Reply:
Hi @jnm2, the change can indeed be made by adding a virtual method to the `TestFilter` class instead, so that no changes are made to the interface.

I have implemented the change and can update the PR if you would like, but as I am new to PR's could you tell me if I should just push a new commit to the branch? I'm unsure how it plays together with all the references to the code in these comments.

Reply:
If you just push, then the PR is updated. But if you are reversing a lot of changes, it's cleaner to first do a reset to the prior HEAD followed by a forced push. (If that's new to you, the simpler way us OK.)

Reply:
I did a force push. I'm comfortable enough with Git. It is mostly the Github part that's new to me :)

Reply:
I did not realize that this was waiting on me because I was supposed to have pressed resolve. Sorry :(